src/key.rs

```diff
@@ -0,0 +1,378 @@
+            "w3": "21888242871839275217838484774961031246154997185409878258781734729429964517155",
+            "w4": "21888242871839275217838484774961031246007050428528088939761107053157389710902",
+            "w8": "19540430494807482326159819597004422086093766032135589407132600596362845576832",
+            "wr": "18200100796661656210024324131237448517259556535315737226009542456080026430510",
+            "X_2": [
+            [
+            "21831381940315734285607113342023901060522397560371972897001948545212302161822",
+            "17231025384763736816414546592865244497437017442647097510447326538965263639101"
+            ],
+            [
+            "2388026358213174446665280700919698872609886601280537296205114254867301080648",
+            "11507326595632554467052522095592665270651932854513688777769618397986436103170"
+            ],
+            [
+            "1",
+            "0"
+            ]
+            ],
+            "C0": [
+            "7436841426934271843999872946312645822871802402068881571108027575346498207286",
+            "18448034242258174646222819724328439025708531082946938915005051387020977719791",
+            "1"
+            ]
+        }
+        "#;
+            let vk: VerificationKey = serde_json::from_str(json).unwrap();
+
+            assert_eq!(VerificationKey::default(), vk);
+        }
+
+        #[test]
+        fn serialize_deserialize_default_key() {
+            let vk = VerificationKey::default();
+            let json = serde_json::to_string(&vk).unwrap();
+            let other = serde_json::from_str(&json).unwrap();
+
+            assert_eq!(vk, other);
+        }
+
+        #[test]
+        fn serialize_deserialize_in_a_non_human_readable_format() {
+            let vk = VerificationKey::default();
+            let mut buffer = [0_u8; 600];
+            ciborium::into_writer(&vk, buffer.as_mut_slice()).unwrap();
+            let other = ciborium::from_reader(buffer.as_slice()).unwrap();
+            assert_eq!(vk, other);
+        }
+
+        #[test]
+        #[should_panic(expected = "Invalid G1 point")]
+        fn raise_error_if_try_to_deserialize_an_invalid_g1_point() {
+            let json = r#"["1", "2", "3"]"#;
+            #[derive(Deserialize)]
+            #[allow(dead_code)]
+            struct Test(#[cfg_attr(feature = "serde", serde(with = "super::g1"))] G1);
+
+            serde_json::from_str::<Test>(json).unwrap();
+        }
+
+        #[test]
+        #[should_panic(expected = "Invalid G2 point")]
+        fn raise_error_if_try_to_deserialize_an_invalid_g2_point() {
+            let json = r#"[["1", "2"], ["3", "4"], ["5", "6"]]"#;
+            #[derive(Deserialize)]
+            #[allow(dead_code)]
+            struct Test(#[cfg_attr(feature = "serde", serde(with = "super::g2"))] G2);
+
+            serde_json::from_str::<Test>(json).unwrap();
+        }
+    }
+}
+
+impl Default for VerificationKey {
+    fn default() -> Self {
```

Comment:
Is a default `VerificationKey` needed/desired?

Reply:
It's just useful to handle the adoption on NH-core smoothing. I'll remove it for the `1.0.0` version